editor/src/clj/editor/camera_editor.clj

```diff
@@ -98,6 +114,223 @@
+        ;; far vertices
+        far-v0 (gen-outline-vertex (.x far-p0) (.y far-p0) (.z far-p0) (.w far-p0) cr cg cb)
+        far-v1 (gen-outline-vertex (.x far-p1) (.y far-p1) (.z far-p1) (.w far-p1) cr cg cb)
+        far-v2 (gen-outline-vertex (.x far-p2) (.y far-p2) (.z far-p2) (.w far-p2) cr cg cb)
+        far-v3 (gen-outline-vertex (.x far-p3) (.y far-p3) (.z far-p3) (.w far-p3) cr cg cb)
+        ;; near vertices
+        near-v0 (gen-outline-vertex (.x near-p0) (.y near-p0) (.z near-p0) (.w near-p0) cr cg cb)
+        near-v1 (gen-outline-vertex (.x near-p1) (.y near-p1) (.z near-p1) (.w near-p1) cr cg cb)
+        near-v2 (gen-outline-vertex (.x near-p2) (.y near-p2) (.z near-p2) (.w near-p2) cr cg cb)
+        near-v3 (gen-outline-vertex (.x near-p3) (.y near-p3) (.z near-p3) (.w near-p3) cr cg cb)
+
+        cr-less (* cr 0.75)
+        cg-less (* cg 0.75)
+        cb-less (* cb 0.75)
+
+        ;; camera vertices from center to near
+        camera-near-v0 (gen-outline-vertex (.x near-p0) (.y near-p0) (.z near-p0) (.w near-p0) cr-less cg-less cb-less)
+        camera-near-v1 (gen-outline-vertex (.x near-p1) (.y near-p1) (.z near-p1) (.w near-p1) cr-less cg-less cb-less)
+        camera-near-v2 (gen-outline-vertex (.x near-p2) (.y near-p2) (.z near-p2) (.w near-p2) cr-less cg-less cb-less)
+        camera-near-v3 (gen-outline-vertex (.x near-p3) (.y near-p3) (.z near-p3) (.w near-p3) cr-less cg-less cb-less)
+
+        ;; camera vertex for center
+        camera-v (gen-outline-vertex (.x camera-pos) (.y camera-pos) (.z camera-pos) 1 cr-less cg-less cb-less)]
+    ;; Add camera mesh vertices
+    (conj-camera-mesh-vertices! vbuf inv-view cr cg cb)
+    (-> vbuf
+        ;; Add square for near plane
+        (conj! near-v0) (conj! near-v1) (conj! near-v1) (conj! near-v2) (conj! near-v2) (conj! near-v3) (conj! near-v3) (conj! near-v0)
+        ;; Add square for far plane
+        (conj! far-v0) (conj! far-v1) (conj! far-v1) (conj! far-v2) (conj! far-v2) (conj! far-v3) (conj! far-v3) (conj! far-v0)
+        ;; Add lines for frustum from near to far plane corners
+        (conj! near-v0) (conj! far-v0)
+        (conj! near-v1) (conj! far-v1)
+        (conj! near-v2) (conj! far-v2)
+        (conj! near-v3) (conj! far-v3)
+        ;; Add lines from camera position to near plane
+        (conj! camera-v) (conj! camera-near-v0)
+        (conj! camera-v) (conj! camera-near-v1)
+        (conj! camera-v) (conj! camera-near-v2)
+        (conj! camera-v) (conj! camera-near-v3))))
+
+(vtx/defvertex color-vtx
+  (vec3 position)
+  (vec4 color))
+
+(defn- camera-view-matrix [^Vector3d position ^Quat4d rotation]
+  (let [m (Matrix4d.)
+        p (Vector3d. position)]
+    (.setIdentity m)
+    (.set m rotation)
+    (.transpose m)
+    (.transform m p)
+    (.negate p)
+    (.setColumn m 3 (.x p) (.y p) (.z p) 1.0)
+    m))
+
+(defn- camera-perspective-projection-matrix [near far fov]
```

Comment:
I think these are copied from somewhere, maybe I should use wherever this came from instead